networkit/cpp/distance/DynDijkstra.cpp

```diff
@@ -38,65 +92,131 @@ void DynDijkstra::update(GraphEvent e) {
     updateBatch(batch);
 }
 
-void DynDijkstra::updateBatch(const std::vector<GraphEvent>& batch) {
+void DynDijkstra::updateBatch(const std::vector<GraphEvent> &batch) {
     mod = false;
     // priority queue with distance-node pairs
     Aux::PrioQueue<edgeweight, node> Q(G->upperNodeIdBound());
```

Comment:
If you have time, using a heap here would be more efficient.

Reply:
will do.

Reply:
So, since the prioqueue handles values that are not (immediately) written to the distance array could save the contents of the queue in its own array (would mean keeping a second array in which most of its values are probably not used in bigger graphs) in a hashmap and use an additional tlx heap to sort(which I doubt it would be faster).

Reply:
Yes, the heap also avoids duplicating the keys. You can create the heap using the comparator `Aux::LessInVector` to compare distances.